node/src/components/contract_runtime/operations.rs

```diff
@@ -112,6 +112,8 @@ pub fn execute_finalized_block(
                 finalized_block.timestamp().millis(),
                 finalized_block.era_id().successor(),
             )?;
+            // Flush if we ran commit_step
+            engine_state.flush_environment()?;
```

Comment:
Wouldn't it be equivalent if we called `flush_environment()?` once, after this code block? We flush after deploy execution a couple of lines above this (before we enter into the auction code block).

Reply:
Done